x/precisebank/keeper/burn.go

```diff
@@ -52,12 +53,23 @@ func (k Keeper) BurnCoins(ctx sdk.Context, moduleName string, amt sdk.Coins) err
 		}
 	}
 
-	// No more processing required if no ExtendedCoinDenom
-	if extendedAmount.IsZero() {
+	// Only burn extended coin if the amount is positive
+	if extendedAmount.IsPositive() {
+		if err := k.burnExtendedCoin(ctx, moduleName, extendedAmount); err != nil {
+			return err
+		}
+	}
+
+	fullEmissionCoin := getFullExtendedValueAmount(amt)
+	if fullEmissionCoin.IsZero() {
```

Comment:
Yeah I see slightly more performant when there's zero amount? Though I think it's clearer with it below since it's only meant for emitting events, becomes a bit more ambiguous to me if its before the burnExtendedCoins() call